openerp/osv/expression.py

```diff
@@ -869,8 +869,7 @@ def push_result(leaf):
             # Making search easier when there is a left operand as column.o2m or column.m2m
             elif len(path) > 1 and column and column._type in ['many2many', 'one2many']:
                 right_ids = comodel.search(cr, uid, [(path[1], operator, right)], context=context)
-                table_ids = model.search(cr, uid, [(path[0], 'in', right_ids)], context=dict(context, active_test=False))
-                leaf.leaf = ('id', 'in', table_ids)
+                leaf.leaf = (path[0], 'in', right_ids)
```

Comment:
@sbidoul it's indeed the same. However, merging cases is a refactoring that should not be done in a stable version. Moreover, the cases are not the same in 9.0, so that merging will make the forward-porting process more complicated.